### ISSUE TITLE
[feature] Add pip-system-certs to python dependencies

### ISSUE THREAD

**Issue by daltonv** [conan-io/docs#3123]
### What is your suggestion?

As conan seems to be widely used in corporate networks it seems like it would be a good idea to add [pip-system-certs](https://pypi.org/project/pip-system-certs/) to the dependencies. 

This should help anybody who has corporate self-signed certificates due to ssl inspection policies, as it forces python `requests` to use the system certificate store as well as the `certifi` store. The self signed cert aught to be on the system store, so no extra work is needed by the dev to get `conan install` to work. 

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by puetzk** [comment]
Looks like [pip settled on implementing a `--use-feature=truststore`](https://discuss.python.org/t/announcement-pip-22-2-release/17543) starting from 22.2.

So maybe conan could/should grow a `core.net.http:truststore=true` conf option, or perhaps special-case `core.net.http:cacert_path=truststore` as a "magic" path for referring to the OS one, or something like that.

**Comment by puetzk** [comment]
Also some interesting discussion (from the author of pip-system-certs), about how `truststore.inject_into_ssl` compares to his current implementation at https://github.com/sethmlarson/truststore/issues/75#issuecomment-1605459398

definitely sounds like he thinks they did it better, and the remaining difference is just whether it's done by the app or via a .pth hook for the whole python installation/venv. Which is moot if it was going to be something controlled by the app (e.g. pip's feature flag or a similar conf for conan).

**Comment by memsharded** [comment]
This seems interesting.
https://github.com/psf/requests/issues/2966#issuecomment-1845736477 and truststore itself describe how it can be used to inject into ssl, so ``requests`` doesn't really need any modification to use it.

There could be some problems, like for example the library only works with Python 3.10 and certain system apis, which means it can be almost impossible to make it available by default in Conan, so there should be some kind of opt-in.

**Comment by puetzk** [comment]
Yeah, you'd just make it a conditional requirement in requirements.txt : `truststore >= 0.6; python_version >= '3.10'`

Then gate the `import truststore` + `truststore.inject_into_ssl()`, behind something in global.conf. Maybe duplicate the version check to report a nicer "truststure requires python 3.10" error message if someone tries to turn it on but lacks the prerequisites (or just let the import fail...)

**Comment by puetzk** [comment]
Nevermind, truststore itself already does that with a custom `ImportError`: https://github.com/sethmlarson/truststore/blob/178fea29c3fb5b14a62deeb112ed3e3d81e95157/src/truststore/__init__.py#L5-L6

**Comment by puetzk** [comment]
The dependency on >= 3.10 is (unfortunately) quite real: https://github.com/sethmlarson/truststore/issues/107#issuecomment-1605482423, that's when the `ssl` module was extended with the APIs truststore can use to hook itself in.

**Comment by memsharded** [comment]
> Yeah, you'd just make it a conditional requirement in requirements.txt : truststore >= 0.6; python_version >= '3.10'

I am still a bit concerned about adding this kind of dependency to Conan for every Conan users with Python 3.10, that could have issues with system requirements for example, so I'd be more inclined to make it opt-in initially protecting the ``import`` in Conan codebase only when enabled or something like that. It is still early stage, the repo has just 125 stars, still depends too much on a single person, etc, and we have had painful experiences with this kind of dependency in the past.

**Comment by puetzk** [comment]
Fair. I was looking at adoption within cpython,more so than github stars. [pip >= 23.3 has it always-installed](https://github.com/pypa/pip/commit/9ee4b8ce36ce3f0f57615db017334a43a97d2dea) (though not automatically imported/used). Though pip of course has vendored it rather than listed in requirements.txt, since pip itself can't rely on getting things via pip. And this version of pip which bundles a vendored truststore is what's being built for the now in the cpython [3.11](https://github.com/python/cpython/pull/112719) and [3.12](https://github.com/python/cpython/pull/112718) branches. Though this has only happened in the last few months, of course, but "`truststore` being included in ensurepip and thus in the default installation of cpython" seems like a pretty firm vote of "what python and requests will be doing".

Pip **hasn't** make `--use-feature=truststore` the default yet, though I think that's coming eventually; see https://github.com/pypa/pip/pull/11647. My guess is that won't happen until after python 3.9 goes out of support next year, since a pip default that didn't work on all supported versions of python like a thing they'd avoid.

So I was leaning toward that same practice - installed by default (to avoid the chicken-and-egg problem that you can't install it because your SSL settings don't work without without it, but keeping actual use opt-in for now.

**Comment by Nachicle** [comment]
Hi! I'd like to add to this conversation that using certificates signed by authorities like DigiCert doesn't work properly too, it's not just a self-signed certificates issue.

I had to install pip-system-certs to get rid of SSL verification errors.
